tools/functional-tester/etcd-tester/cluster.go

```diff
@@ -266,12 +266,23 @@ func setHealthKey(us []string) error {
 		if err != nil {
 			return fmt.Errorf("%v (%s)", err, u)
 		}
-		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
 		kvc := pb.NewKVClient(conn)
-		_, err = kvc.Put(ctx, &pb.PutRequest{Key: []byte("health"), Value: []byte("good")})
-		cancel()
+		for i := 0; i < 5; i++ {
+			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+			_, err = kvc.Put(ctx, &pb.PutRequest{Key: []byte("health"), Value: []byte("good")})
+			cancel()
+			if grpc.ErrorDesc(err) == context.DeadlineExceeded.Error() {
+				log.Printf("etcd-tester: kvc.Put retrying with %v", err)
+				time.Sleep(time.Second)
+				continue
+			}
+			if err == nil {
```

Comment:
check the error == nil case before the deadline case. usually we want to handle simple and most common case first.